Assets/Scripts/Tracker/BaseTracker.cs

```diff
@@ -0,0 +1,100 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections;
+using System.Collections.Generic;
+using Logger = Mediapipe.Unity.Logger;  // Disambiguation with UnityEngine.Logger
+using Mediapipe;
+using Mediapipe.Unity;
+using Mediapipe.Unity.Holistic;
+using UnityEngine;
+
+public abstract class BaseTracker : MonoBehaviour {
+  private const string _TAG = nameof(BaseTracker);
+  public enum InferenceMode {
+    CPU,
+    GPU,
+  }
+  [Tooltip("Inference mode.")]
+  [SerializeField] private InferenceMode _preferableInferenceMode;
+  [Tooltip("Reference to the internal texture frame pool.")]
+  [SerializeField] private TextureFramePool _textureFramePool;
+  [SerializeField] private Texture _sourceTexture;
+  [Tooltip("Whether to flip the source image horizontally before processing.")]
+  [SerializeField] private bool _hFlip;
+  [Tooltip("Whether to flip the source image vertically before processing.")]
+  [SerializeField] private bool _vFlip;
+  [Tooltip("Angles to rotate the source image.")]
+  [SerializeField] private int _rotation;
+  protected HolisticTrackingGraph _graphRunner;
+  private Coroutine _coroutine;
+  private InferenceMode _inferenceMode;
+
+  // <summary>Attaches any event handler for callbacks. </summary>
+  public abstract void AddEventHandler();
+
+  public IEnumerator Start() {
+    _graphRunner = GetComponent<HolisticTrackingGraph>();
+    AssetLoader.Provide(new StreamingAssetsResourceManager());
+
+    Logger.LogInfo(_TAG, "Starting mediapipe");
+    DecideInferenceMode();
+    if (_inferenceMode == InferenceMode.GPU) {
+      Logger.LogInfo(_TAG, "Initializing GPU resources...");
+      yield return GpuManager.Initialize();
+    }
+    var graphInitRequest = _graphRunner.WaitForInit(RunningMode.Async);
+    _textureFramePool.ResizeTexture(_sourceTexture.width, _sourceTexture.height, TextureFormat.RGBA32);
+    yield return graphInitRequest;
+    if (graphInitRequest.isError) {
+      Debug.Log(graphInitRequest.error);
+      yield break;
+    }
+    AddEventHandler();
+    Logger.LogInfo(_TAG, "Graph Runner Init!");
+    SidePacket sidePacket = _graphRunner.BuildSidePacket(_rotation, _hFlip, _vFlip);
+    _graphRunner.StartRun(sidePacket);
+    Logger.LogInfo(_TAG, "Graph Runner started in async mode!");
+
+    _coroutine = StartCoroutine(ProcessImage(_sourceTexture));
+  }
+
+  public void OnDestroy() {
+    if (_coroutine != null) StopCoroutine(_coroutine);
+  }
+
+  private IEnumerator ProcessImage(Texture image) {
+    Logger.LogVerbose(_TAG, "Process image");
+    while (true) {
+      if (!_textureFramePool.TryGetTextureFrame(out var textureFrame)) {
+        yield return new WaitForEndOfFrame();
+        continue;
+      }
+      textureFrame.ReadTextureFromOnCPU(image);
+      _graphRunner.AddTextureFrameToInputStream(textureFrame);
+      yield return new WaitForEndOfFrame();
```

Comment:
Do you want to run each cycle of the while loop exactly at the end of each frame, or just simply run each cycle every frame?

As I understand `yield return null` and `yield return new WaitForEndOfFrame()`  are slightly different about the next running time. See a clear explanation at https://answers.unity.com/questions/755196/yield-return-null-vs-yield-return-waitforendoffram.html

If `yield return new WaitForEndOfFrame()` is the expected behavior (wait until everything is rendered in every frame), probably we need `yield` out before any actual logic is performed in the while loop:

```
while (ture) {
  yield return new WaitForEndOfFrame();
  // Do some things that have to be run in the end of each frame.
}
```

Reply:
The previous code snippet has some post frame rendering logic, which has been cleaned up. Changing to `yield return null;`